java-frontend/src/main/java/org/sonar/java/UCFGJavaVisitor.java

```diff
@@ -93,14 +94,25 @@ public void scanFile(JavaFileScannerContext context) {
   @Override
   public void visitMethod(MethodTree tree) {
     super.visitMethod(tree);
-    if (tree.block() != null) {
+    Stream<Type> types;
```

Comment:
What about using a simple `List<Type>` first, being directly `tree.symbol().parameterTypes();`, and then only if it's a constructor, simply add the return type to a (new) list including parameters. Then you can call `types.stream()` only once before calling `noneMatch`.